src/Sylius/Behat/Context/Setup/PromotionContext.php

```diff
@@ -142,24 +138,20 @@ public function itGivesFixedDiscountToEveryOrder(PromotionInterface $promotion,
      */
     public function itGivesPercentageDiscountToEveryOrder(PromotionInterface $promotion, $discount)
     {
-        $action = $this->actionFactory->createPercentageDiscount($discount);
-        $promotion->addAction($action);
-
-        $this->objectManager->flush();
+        $this->createPercentagePromotion($promotion, $discount);
     }
 
     /**
      * @Given /^([^"]+) gives ("[^"]+") fixed discount to every order with quantity at least ([^"]+)$/
```

Comment:
I was thinking about it, but there could also be ``the promotion`` and ``that promotion``... It would look nasty, perhaps we should just assume, that step is correctly written and implemented? :) If there be a string which cannot be parsed by any of transformers, it will throw exception anyway.